### DISCUSSION TITLE
[Idea] Beta channel on F-Droid

### DISCUSSION THREAD

**Discussion by StefanoWoerner** [TechbeeAT/jtxBoard#435] [Ideas]
Quite a few people interested in FOSS development prefer installing apps through F-Droid. Since those FOSS-interested people are the most likely to give feedback, it may be worth it to consider adding a beta channel to F-Droid.

I have no idea how much effort maintaining an additional F-Droid package is and if it would be really worth it. Just wanted to put this idea out there.

**Comment by patrickunterwegs** [comment]
Hey, so basically an app has a package name which acts as an identifier for the OS and appstores. That's why you might see the same app in multiple different stores as installed. 
f-Droid does not have a beta channel (unlike Google Play), so the only way to offer betas on f-Droid would be by providing either all versions, independent of beta status or not, or to make a version with a dedicated package name, which would be like another app that needs to be considered in DAVx5 as well. Both ways are just workarounds for the incapability of f-Droid to provide a beta channel. 

In my opinion this is something that should be addressed and solved on f-Droid side. Everything else would just be a workaround...

**Reply by StefanoWoerner** [reply]
That makes a lot of sense. My idea was creating a separate package, similar to "Nextcloud" and "Nextcloud dev", but I didn't consider DAVx5 having to account for an additional app.

**Comment by rogercreagh** [comment]
Since you are now making the apk's available on here perhaps a better solution is to expect those capable of contributing to a beta channel to be able to download and install apk's directlly themselves.

The only reason I don't do this myself is that I am now using jtxBoard as part of my every day life and am loath to risk a beta bug causing data loss. I might consider it on an alternative device but currently I am using both tablet and phone for live daily use. 

So my vote would be to make beta apk's available here and not bother with beta versions in any store. I'd rather keep the sotres (whether Fdroid which I use,Google, or Amazon) as clean release versions so they can auto-update without risk. Having beta releases in the same channel sounds very risky.

**Reply by patrickunterwegs** [reply]
Yes, I agree with you. 
I am publishing frequently on the pre-release channel of GitHub which I also use for my internal testing, so those pre-releases should only be used if someone can accept that a change might have breaking consequences.
I'm using the standard-release-channel only after changes were stable on the Google Play Beta channel, so those should be safe.

**Comment by AeliusSaionji** [comment]
You could host your own fdroid repository for easier delivery of the apks you also put on github pre-release?

or perhaps contact izzyondroid to have them host the beta versions, which is (sortof) what fairemail does. https://github.com/M66B/FairEmail/blob/master/FAQ.md#user-content-faq173 (and ctrl f for izzy)

**Reply by patrickunterwegs** [reply]
Hey @AeliusSaionji ! 
Sure a dedicated repository would be an option, but it has become very easy to use the apk from GitHub with tools like Obtainium, if somebody really would like to be a beta-user, there is this easy way to just take the apk, so for now a dedicated repository is not something that I would follow up for now 😉
